src/app/common/services/course.service.ts

```diff
@@ -14,9 +13,13 @@ import { CourseDeleteDialogComponent } from '../dialog/course-delete-dialog/cour
 })
 export class CourseService {
 
-  protected courses: BehaviorSubject<ICourse[]> = new BehaviorSubject([]);
+  public courses: ICourse[];
```

Comment:
If you have courses in observable there is no need to store courses as an array.